blob/rollup.config.js

```diff
@@ -0,0 +1,61 @@
+import nodeResolve from 'rollup-plugin-node-resolve';
+import { uglify } from 'rollup-plugin-uglify';
+import replace from 'rollup-plugin-replace';
+import commonjs from 'rollup-plugin-commonjs';
+import shim from 'rollup-plugin-shim';
+
+export default [
+  {
+    external: ['ms-rest-js', 'crypto', 'fs', 'events', 'os'],
+    input: 'dist-esm/lib/index.js',
+    output: {
+      file: 'dist/index.js',
+      format: 'cjs',
+      sourcemap: true
+    },
+    plugins: [nodeResolve({ module: true }), uglify()]
+  },
+  {
+    external: ['ms-rest-js'],
+    input: 'dist-esm/lib/index.browser.js',
+    output: {
+      globals: {
+        'ms-rest-js': 'msRest'
+      },
+      file: 'dist/index.browser.js',
+      format: 'umd',
+      name: 'azblob',
+      sourcemap: true
+    },
+    plugins: [
+      replace({
+        delimiters: ['', ''],
+        values: {
+          // replace dynamic checks with if (false) since this is for
+          // browser only. Rollup's dead code elimination will remove
+          // any code guarded by if (isNode) { ... }
+          'if (isNode)': 'if (false)'
+        }
+      }),
+      // os is not used by the browser bundle, so just shim it
+      shim({
+        os: `
+            export const type = 1;
+            export const release = 1;
+          `
+      }),
+      nodeResolve({
+        module: true,
+        browser: true,
+        preferBuiltins: false
+      }),
+      commonjs({
+        namedExports: {
+          events: ['EventEmitter'],
+          assert: ['ok', 'deepEqual', 'equal', 'fail', 'deepStrictEqual']
```

Comment:
Why do we want assert to be part of our browser bundle? assert takes about 15KB in the bundle before minified.

Reply:
This is just a mapping of named exports should they need to be included in the bundle. In this case, assert isn't included in the non-test bundle. I would have preferred to put this in the test rollup config but I couldn't find an easy way to configure a specific plugin differently in the test config.

Reply:
Got it